agent/exec/plugin/adapter.go

```diff
@@ -0,0 +1,48 @@
+package plugin
+
+import (
+	"github.com/docker/docker/api/types"
+	engineapi "github.com/docker/docker/client"
+	"github.com/docker/swarmkit/api"
+	"golang.org/x/net/context"
+)
+
+type pluginAdapter struct {
+	client engineapi.APIClient
+	plugin *pluginConfig
+}
+
+func newPluginAdapter(client engineapi.APIClient, task *api.Task) (*pluginAdapter, error) {
+	plgn, err := newPluginConfig(task)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pluginAdapter{
+		client: client,
+		plugin: plgn,
+	}, nil
+}
+
+func (p *pluginConfig) pluginInstallOptions() types.PluginInstallOptions {
+	return types.PluginInstallOptions{
+		Disabled:             true,
```

Comment:
I agree, but the controller has 2 methods `Prepare` and `Start` I think Prepare should only install and start actually enable it don't you think ?